### PR TITLE
Improve GUI error handling

### PR DESCRIPTION
**Pull request by drew2a**
This PR improves GUI error handling by adding:
1. A user identification based on the user's public key
1. Breadcrumbs for QWidget and QDialog 

An example of an issue with these improvements: https://sentry.tribler.org/organizations/tribler/issues/265

### CODE REVIEW

File: src/tribler-gui/tribler_gui/event_request_manager.py [hunk 1]
```diff
@@ -56,16 +57,17 @@ def __init__(self, api_port, api_key):
...
-            NTFY.TRIBLER_STARTED.value: lambda data: self.tribler_started.emit(data["version"]),
+            NTFY.EVENTS_START.value: lambda _: None,
+            NTFY.TRIBLER_STARTED.value: self.tribler_started_event,
         }
 
-    def events_start_received(self, event_dict):
-        if event_dict["tribler_started"]:
-            self.tribler_started.emit(event_dict["version"])
+    def tribler_started_event(self, event_dict):
+        self.tribler_started.emit(event_dict["version"], event_dict["public_key"])
```
Comment (drew2a) [inline src/tribler-gui/tribler_gui/event_request_manager.py:65]: I've changed the event in which `tribler_starter` is emitted. 
@ichorid please, verify the correctness.

File: src/tribler-core/tribler_core/restapi/events_endpoint.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def passthrough(x):
     # Remote GigaChannel search results were received by Tribler. Contains received entries.
     NTFY.REMOTE_QUERY_RESULTS: passthrough,
     # An indicator that Tribler has completed the startup procedure and is ready to use.
-    NTFY.TRIBLER_STARTED: lambda *_: {"version": version_id},
+    NTFY.TRIBLER_STARTED: lambda public_key: {"version": version_id, "public_key": hexlify(public_key)},
```
Comment (ichorid) [inline src/tribler-core/tribler_core/restapi/events_endpoint.py:48]: The perfect solution to get the key through! :+1:  :clap:

File: src/tribler-gui/tribler_gui/event_request_manager.py (hunk 1 above)
Comment (ichorid) [reply src/tribler-gui/tribler_gui/event_request_manager.py:65]: Actually, the startup procedure is pretty tricky in regards to `EVENTS_START`-`TRIBLER_STARTED` interplay. The tricky part is the upgraded...

To be sure the thing will not break the upgrader, you'll have to test Tribler upgrading the database, say from 7.5->7.6 *and* trying to skip the upgrade.

File: src/tribler-gui/tribler_gui/tribler_window.py [hunk 3]
```diff
@@ -437,10 +437,16 @@ def tray_show_message(self, title, message):
...
-    def on_tribler_started(self, version):
+    def on_tribler_started(self, version, public_key):
         if self.tribler_started:
             logging.warning("Received duplicate Tribler Core started event")
             return
+
+        if public_key:
+            # if public key format will be changed, don't forget to change it
+            # at the core side as well
+            SentryReporter.set_user(public_key.encode('utf-8'))
```
Comment (ichorid) [inline src/tribler-gui/tribler_gui/tribler_window.py:448]: :+1:

File: src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py [hunk 4]
```diff
@@ -31,8 +32,8 @@
 
 widget_form, widget_class = uic.loadUiType(get_ui_file_path('torrents_list.ui'))
 
-
-class ChannelContentsWidget(widget_form, widget_class):
+# pylint: disable=too-many-instance-attributes, too-many-public-methods
```
Comment (ichorid) [inline src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py:35]: Thanks...

File: src/tribler-gui/tribler_gui/event_request_manager.py (hunk 1 above)
Comment (drew2a) [reply src/tribler-gui/tribler_gui/event_request_manager.py:65]: Mmm... I didn't think it is so complex.
Better to revert back then.

Comment (ichorid) [reply src/tribler-gui/tribler_gui/event_request_manager.py:65]: It is only complex in regards to the `EVENTS_START` signal. Please, just test it with the upgraded and see if it works. If it does, keep your changes. I remember trying to remove the `EVENTS_START` signal altogether but failed due to the upgraded shenanigans. Maybe you'll have better luck with it :four_leaf_clover: :wink: 

If you don't want to risk it, you could still keep you `public_key`-passing change and just revert the `EVENTS_START`-related stuff.

Comment (drew2a) [reply src/tribler-gui/tribler_gui/event_request_manager.py:65]: It seems to be working.

**Review by ichorid — APPROVED** [review]
:+1:

File: src/tribler-core/tribler_core/session.py [hunk 5]
```diff
@@ -280,7 +281,11 @@ async def start(self):
         self.get_ports_in_config()
         self.create_state_directory_structure()
         self.init_keypair()
-        SentryReporter.set_user(self.trustchain_keypair.key.pk)
+
+        # we have to represent `user_id` as a string to make it equal to the
+        # `user_id` on the GUI side
+        user_id_str = hexlify(self.trustchain_keypair.key.pk).encode('utf-8')
```
Comment (xoriole) [inline src/tribler-core/tribler_core/session.py:287]: Using `mid` could be more privacy preserving

File: src/tribler-gui/tribler_gui/tribler_window.py (hunk 3 above)
Comment (xoriole) [reply src/tribler-gui/tribler_gui/tribler_window.py:454]: Similarly here, using `mid` could be privacy preserving

**Review by xoriole — APPROVED** [review]

**Review by kozlovsky — APPROVED** [review]

**Pull request merged**